packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java

```diff
@@ -542,7 +539,7 @@ public void uploadAsync(final String fileUriString, final String url, final Map<
         promise.reject("ERR_FILESYSTEM_MISSING_UPLOAD_TYPE", "Missing upload type.", null);
         return;
       }
-      UploadType uploadType = UploadType.fromInteger((Integer) options.get("uploadType"));
+      UploadType uploadType = UploadType.fromInt(((Double) options.get("uploadType")).intValue());
```

Comment:
```suggestion
      UploadType uploadType = UploadType.fromInteger(((Double) options.get("uploadType")).intValue());
```
I'm pretty sure that this mechanism will take care of `int <=> Integer` conversion: https://docs.oracle.com/javase/8/docs/technotes/guides/language/autoboxing.html
But maybe there's any particular reason that you've changed this type to primitive.
You don't have to adjust it here, but with this one you might remove that other connected change 😉

Reply:
I know that autoboxing can convert `int` to `Integer` her, but without it there is a fewer type conversion:
- earlier (`UploadType.fromInteger`)
	`object` => `Double` => `int` => `Integer`
- now 
	`object` => `Double` => `int`